apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -491,86 +491,44 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
-                if (EnableJsonLayout)
                 {
-                    foreach (var jToken in JObject.Parse(RenderLogEvent(Layout, loggingEvent)))
-                    {
-                        jsonStruct.Fields.Add(jToken.Key, _jsonConvertFunction(jToken.Value));
-                    }
-                }
-                else
-                {
-                    jsonStruct.Fields.Add("message", Value.ForString(RenderLogEvent(Layout, loggingEvent)));
-                    
-                    if (ServiceContextName != null)
-                    {
-                        var serviceName = RenderLogEvent(ServiceContextName, loggingEvent);
-                        if (!string.IsNullOrEmpty(serviceName))
-                        {
-                            // Include ServiceContext to allow errors to be automatically forwarded
-                            var serviceVersion = RenderLogEvent(ServiceContextVersion, loggingEvent);
-                            if (string.IsNullOrEmpty(serviceVersion))
-                            {
-                                serviceVersion = "0.0.0.0";
-                            }
-
-                            var serviceContext = new Struct();
-                            jsonStruct.Fields.Add("serviceContext", Value.ForStruct(serviceContext));
-                            serviceContext.Fields.Add("service", Value.ForString(serviceName));
-                            serviceContext.Fields.Add("version", Value.ForString(serviceVersion));
-                        }
-                    }
+                    logEntry.Trace = _traceResourcePath + traceId;
 
-                    var propertiesStruct = new Struct();
-                    jsonStruct.Fields.Add("properties", Value.ForStruct(propertiesStruct));
-
-                    foreach (var combinedProperty in GetAllProperties(loggingEvent).Where(x => !string.IsNullOrEmpty(x.Key)))
+                    var spanId = RenderLogEvent(SpanId, loggingEvent);
+                    if (!string.IsNullOrEmpty(spanId))
                     {
-                        Value jsonValue;
-                        try
-                        {
-                            jsonValue = _jsonConvertFunction(combinedProperty.Value);
-                        }
-                        catch (Exception ex)
-                        {
-                            InternalLogger.Warn(ex,
-                                "GoogleStackdriver(Name={0}): Exception at BuildLogEntry with Key={1}", Name, combinedProperty.Key);
-                            jsonValue = Value.ForString($"<Exception: '{ex.Message}'>");
-                        }
-
-                        propertiesStruct.Fields.Add(combinedProperty.Key, jsonValue);
+                        logEntry.SpanId = spanId;
                     }
+
+                    logEntry.TraceSampled = TraceSampled;
                 }
-                logEntry.JsonPayload = jsonStruct;
+            }
+
+            if (SendJsonPayload)
```

Comment:
Please undo the changes that are not related to #5677. It's totally fine if you want to refactor this code some, but please do so in another PR.

Reply:
Undid refactoring